src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -455,28 +503,39 @@ void activity_classifier::init_train(
     std::string session_id_column_name, variant_type validation_data,
     std::map<std::string, flexible_type> opts)
 {
-
+  gl_sframe train_data;
+  gl_sframe val_data;
   // Begin printing progress.
   // TODO: Make progress printing optional.
   if (variant_is<gl_sframe>(validation_data)) {
```

Comment:
Could we create a helper function that initializes `train_data` and `val_data`? So we can do something like
```
std::tie(train_data, val_data) = init_data(data, validation_data, session_id_column_name);
init_table_printer(!val_data.empty());
```